### PR TITLE
CI: run database tests only #39550

### LINKED ISSUE #39550: CI: have database builds *just* run db tests
The database builds: https://github.com/pandas-dev/pandas/pull/39457/checks?check_run_id=1797892949

take the longest. we should just have this run the database tests only. (so add a marker for the appropriate tests / just run the sql tests entirely).

### LINKED ISSUE #39758: CI: remove the current coverage build, run coverage on multiple builds instead
Currently, we upload one coverage report from `actions-37-cov`. It takes the longest. But Codecov automatically merges multiple reports. See details https://docs.codecov.io/docs/merging-reports. So we could run coverage on multiple shorter builds.

We need to select a set of builds (create new ones if needed), add `-cov` to the name of their environment files (suggested by @jreback ), update their CI configs to upload reports to Codecov, and remove the `actions-37-cov` build if it's no longer required.

### PR DESCRIPTION
**Pull request by fangchenli**
- [x] closes #39550
- [x] closes #39758

### CODE REVIEW

**Comment by jreback** [comment]
hmm maybe its the db - cov build that is a problem. i guess any way to split up these cov builds? (this might be harder, as now we would have multiple uploads to code-cov, but also maybe unavoidable)

**Comment by fangchenli** [comment]
> hmm maybe its the db - cov build that is a problem. i guess any way to split up these cov builds? (this might be harder, as now we would have multiple uploads to code-cov, but also maybe unavoidable)

https://docs.codecov.io/docs/merging-reports Codecov automatically merges multiple reports. So we don't even need a separate build for coverage. We could select a set of builds that include all tests, run coverage, and upload reports for all of them.

**Comment by jreback** [comment]
> > hmm maybe its the db - cov build that is a problem. i guess any way to split up these cov builds? (this might be harder, as now we would have multiple uploads to code-cov, but also maybe unavoidable)
> 
> https://docs.codecov.io/docs/merging-reports Codecov automatically merges multiple reports. So we don't even need a separate build for coverage. We could select a set of builds that include all tests, run coverage, and upload reports for all of them.

that would be great! (but should tag the builds as -cov just to indicate that they are part of the coverage build(s))

**Comment by fangchenli** [comment]
I probably won't have time to work on the coverage stuff in the next couple of days. I opened #39758. So people can take from there.

File: .github/workflows/database.yml [hunk 1]
```diff
@@ -20,94 +20,13 @@ jobs:
...
-        fail_ci_if_error: false
-
-  Linux_py37_cov:
-    runs-on: ubuntu-latest
-    defaults:
-      run:
-        shell: bash -l {0}
+    strategy:
+      matrix:
+        ENV_FILE: [ci/deps/actions-37-locale.yaml, ci/deps/actions-37-cov.yaml]
```
Comment (MarcoGorelli) [inline .github/workflows/database.yml:25]: Hi @fangchenli - in #39758 you wrote

> and remove the actions-37-cov build if it's no longer required.

but here it looks like you're keeping it - is it still required then?

Comment (fangchenli) [reply .github/workflows/database.yml:25]: One of them is testing the minimum support versions of SQL-related packages. So I think it's better to keep both of them. But should rename them to reflect their actual purpose.

**Comment by jreback** [comment]
thanks @fangchenli very nice

**Pull request merged**